packages/ember-environment/lib/index.js

```diff
@@ -66,8 +66,10 @@ ENV.LOG_STACKTRACE_ON_DEPRECATION = defaultTrue(ENV.LOG_STACKTRACE_ON_DEPRECATIO
 */
 ENV.LOG_VERSION = defaultTrue(ENV.LOG_VERSION);
 
-// default false
-ENV.MODEL_FACTORY_INJECTIONS = defaultFalse(ENV.MODEL_FACTORY_INJECTIONS);
+// always false
+Object.defineProperty(ENV, 'MODEL_FACTORY_INJECTIONS', {
```

Comment:
We should issue a deprecation if `MODEL_FACTORY_INJECTIONS in ENV`, and update ember-cli to confirm it isn't specified in the blueprint. 

Then instead of setting it to false here, we just leave it undefined (which would also be seen as falsey by ED's assertion).